tests/test_routing.py

```
@@ -1310,6 +1310,118 @@ async def test_update_cache_writes_marker_with_dim(self, isolated_router_dir, mo
+
+        from src.engine.router import SemanticRouter
+        router = SemanticRouter()
+        assert "deleted_legacy_lawyer" not in router.available_agents
+        assert "universal_agent" in router.available_agents
+
+        result = await router.query_nearest("any query")
+        assert result is not None, (
+            "query_nearest must skip the stale top-1 and return the next "
+            "valid neighbour, not give up on the first stale hit"
+        )
+        decision, distance = result
```

Ruff (0.15.13)

[warning] Unpacked variable `distance` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)